### PR TITLE
Support custom temp directory

### PR DESCRIPTION
**Pull request by cozmo**
Allow passing in a `tmp_dir` which we pass through to `sort` via the `--temporary-directory` option. 

Not sure best way to test this since sort usually cleans up after itself.

### CODE REVIEW

File: lib/index.coffee [hunk 1]
```diff
@@ -26,9 +27,14 @@ module.exports = (keys, options={}) ->
...
       key_opts = _.flatten _([1..keys.length]).map (i) -> ['-k', "#{i},#{i}"]
-      [ '-s'                # stable sort
+      flags = [
+        '-s'                # stable sort
         '-t', options.delim # column separator
-      ].concat key_opts     # which columns to sort by
+      ]
+      flags = flags.concat ["--temporary-directory", options.tmp_dir] if options.tmp_dir
+      flags = flags.concat key_opts # which columns to sort by
+      console.log flags
```
Comment (azylman) [inline lib/index.coffee]: !!

Comment (cozmo) [inline lib/index.coffee]: 😡  Damnit.

File: lib/index.coffee [hunk 2]
```diff
@@ -13,6 +13,7 @@ Understream.mixin JSONParser, 'parseJSON'
 module.exports = (keys, options={}) ->
   _(options).defaults
     delim: '\t'
+    tmp_dir: null
```
Comment (azylman) [inline lib/index.coffee]: Why default `null` as opposed to just no default?

Comment (cozmo) [inline lib/index.coffee]: Mostly to make it clear this was an option since we don't really have docs. I can remove if we feel that's better.

**Comment by cozmo** [comment]
Offline LGTM from @azylman.

**Pull request merged**